REGRESSION (253593@main): [ macOS ] inspector/canvas/updateShader-webgl.html is a consistent failure

kkinnunen-apple opened the pull request:
#### c984b72df669166a4d5def7ceaf8cb7bafb772c3
<pre>
REGRESSION (253593@main): [ macOS ] inspector/canvas/updateShader-webgl.html is a consistent failure
<a href="https://bugs.webkit.org/show_bug.cgi?id=244159">https://bugs.webkit.org/show_bug.cgi?id=244159</a>
rdar://problem/98926428

Reviewed by Devin Rousso.

Shader compilation status cannot be queried after invoking the compilation,
because this will block otherwise asynchronous method. This means we cannot
print the ad hoc console message on all compile failures. The commit
253593@main removed the message for all WebGL shader compiles.

Add the error message back when user edits the shader with inspector.

* LayoutTests/inspector/canvas/updateShader-webgl-expected.txt:
* LayoutTests/platform/mac/TestExpectations:

Canonical link: <a href="https://commits.webkit.org/253902@main">https://commits.webkit.org/253902@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/4fe5a3ecf495313a25e08523fb57ffed35b03e20

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/87427 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/69/builds/31516 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/18233 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/96655 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/150038 "Built successfully and passed tests") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/91402 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/64/builds/29881 "Built successfully") | [✅ 🛠 mac-debug](https://ews-build.webkit.org/#/builders/71/builds/26065 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/79546 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/91426 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/93045 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/68/builds/24141 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/74213 "Built successfully") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/23979 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/79109 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/79335 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/66997 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/66/builds/27591 "Built successfully") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/13167 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/67/builds/27545 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/14183 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/2742 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/63/builds/29233 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/60/builds/37046 "Passed tests") | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/65/builds/29160 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/33461 "Passed tests") | | 
<!--EWS-Status-Bubble-End-->

dcrousso commented on Source/WebCore/inspector/InspectorShaderProgram.cpp:
```diff
@@ -124,9 +124,19 @@ bool InspectorShaderProgram::updateShader(Inspector::Protocol::Canvas::ShaderTyp
+                                auto errors = contextWebGLBase.getShaderInfoLog(*shader);
+                                auto* scriptContext = m_canvas.scriptExecutionContext();
+                                for (auto error : StringView(errors).split('\n')) {
+                                    auto message = makeString("Error compiling edited WebGL shader: "_s, error);
```
Can we have this just be
```
makeString("WebGL: "_s, error);
```
so that it matches the old error string (and therefore could make the job of identifying WebGL errors easier on the frontend)?

dcrousso reviewed (approved):
nice!  thanks for fixing :)

kkinnunen-apple committed:
REGRESSION (253593@main): [ macOS ] inspector/canvas/updateShader-webgl.html is a consistent failure
https://bugs.webkit.org/show_bug.cgi?id=244159
rdar://problem/98926428

Reviewed by Devin Rousso.

Shader compilation status cannot be queried after invoking the compilation,
because this will block otherwise asynchronous method. This means we cannot
print the ad hoc console message on all compile failures. The commit
253593@main removed the message for all WebGL shader compiles.

Add the error message back when user edits the shader with inspector.

* LayoutTests/inspector/canvas/updateShader-webgl-expected.txt:
* LayoutTests/platform/mac/TestExpectations:

Canonical link: https://commits.webkit.org/253902@main
